server/src/Application/Services/FlashcardImporter/FlashcardImporter.cs

```diff
@@ -39,24 +39,65 @@ public FlashcardImporter(ValidationHandler validationHandler)
         int flashcardSetId)
     {
         var importedFlashcards = new BlockingCollection<CreateFlashcardDto>();
+        var batches = PartitionBatch(lines, 3); 
+        var threadList = new List<Thread>();
 
-        Parallel.ForEach(lines, line =>
+        foreach (var batch in batches)
         {
-            var values = line.Split(',');
-            if (values.Length != 2)
+            var thread = new Thread(() =>
             {
-                throw new IncorrectArgumentCountException("FileContent");
-            }
+                foreach (var line in batch)
+                {
+                    var values = line.Split(',');
+                    if (values.Length != 2)
+                    {
+                        throw new IncorrectArgumentCountException("FileContent");
+                    }
+
+                    var front = values[0];
+                    var back = values[1];
+                    var createFlashcardDto = new CreateFlashcardDto(flashcardSetId, front, back);
 
-            var front = values[0];
-            var back = values[1];
-            var createFlashcardDto = new CreateFlashcardDto(flashcardSetId, front, back);
+                    importedFlashcards.Add(createFlashcardDto);
+                }
+            });
 
-            importedFlashcards.Add(createFlashcardDto);
-        });
+            threadList.Add(thread);
+            thread.Start();
+        }
+
+        foreach (var thread in threadList)
+        {
+            thread.Join();
+        }
 
         importedFlashcards.CompleteAdding();
 
         return importedFlashcards.GetConsumingEnumerable().ToList();
     }
+
+    private IEnumerable<IEnumerable<string>> PartitionBatch(IEnumerable<string> source, int numberOfBatches)
+    {
+        var batch = new List<string>();
+        int batchSize = source.Count() / numberOfBatches;
+        int currentBatchSize = 0;
+
+        foreach (var item in source)
```

Comment:
I think you can use Skip and Take methods https://stackoverflow.com/questions/18986129/how-can-i-split-an-array-into-n-parts